x-pack/plugins/security_solution/public/common/components/autocomplete/helpers.ts

```diff
@@ -30,29 +30,33 @@ export const getOperators = (field: IFieldType | undefined): OperatorOption[] =>
   }
 };
 
-export function validateParams(params: string | undefined, type: string) {
+export const validateParams = (
+  params: string | undefined,
+  field: IFieldType | undefined
+): boolean => {
   // Box would show error state if empty otherwise
   if (params == null || params === '') {
     return true;
   }
 
-  switch (type) {
-    case 'date':
-      const moment = dateMath.parse(params);
-      return Boolean(moment && moment.isValid());
-    case 'ip':
-      try {
-        return Boolean(new Ipv4Address(params));
-      } catch (e) {
-        return false;
-      }
-    case 'number':
-      const val = parseFloat(params);
-      return typeof val === 'number' && !isNaN(val);
-    default:
-      return true;
-  }
-}
+  const types = field != null && field.esTypes != null ? field.esTypes : [];
+
+  return types.reduce<boolean>((acc, type) => {
+    switch (type) {
+      case 'date':
+        const moment = dateMath.parse(params);
+        return Boolean(moment && moment.isValid());
```

Comment:
nit, can you just do: `return moment && moment.isValid()` ? Won't that work as well without introducing the capital boolean boxing here?

Reply:
I tried removing it and the test yelled at me saying expected `boolean` not `string`. Leaving for now.